torchinductor/ir.py

```diff
@@ -1237,15 +1240,44 @@ def __init__(
         )
         assert getattr(torch.ops.aten, kernel.__name__) is kernel
         self.kernel = f"aten.{kernel.__name__}"
+        self.unflatten_args = unflatten_args
+
+    def codegen_args(self):
+        @dataclasses.dataclass
+        class Shim:
+            ref: Any
+
+            def __repr__(self):
+                return self.ref
+
+        tensor_args = [Shim(x.codegen_reference()) for x in self.inputs]
+        constant_args = [Shim(repr(x)) for x in self.constant_args]
+        return list(map(repr, self.unflatten_args(tensor_args, constant_args)))
 
     @classmethod
     def create(cls, kernel, *args):
```

Comment:
What is this logic for handling? Is this assuming that we might take in FX graphs where the inputs is not a flat list of tensors?

Reply:
_cudnn_rnn takes a list of tensors as args.  This is just to support that.